File: translation_output/gpt-4/codenet/python/java/p02390.java
Here is the equivalent Java code for the Python code you have provided:

```java
import java.util.Scanner;

public class TimeConversion {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of seconds: ");
        int ss = scanner.nextInt();
        
        int hour = ss / (60 * 60);
        int second = ss % (60 * 60);
        int minute = second / 60;
        second = second % 60;
        
        System.out.printf("%d:%d:%d\n", hour, minute, second);
        
        scanner.close();
    }
}
```

This Java program will take an input of seconds from the user, convert it to hours, minutes, and seconds, and then print out the result in the specified format (HH:MM:SS). Just like in the Python code, the `Scanner` is used for user input, and `System.out.printf` is used for formatted printing, which is similar to the `format` method in Python.

File: translation_output/gpt-4/codenet/java/python/p02390.py
Here's the translated Java code to Python:

```python
def main():
    S = int(input())
    h = S // 3600
    hi = S % 3600
    m = hi // 60
    s = S % 60
    print(f"{h}:{m}:{s}")

if __name__ == "__main__":
    main()
```

The Python code uses standard input to receive an integer like the Java code. It then performs the same calculations for hours, minutes, and seconds, and prints them in the same format. Python's string formatting (f-string) is used to print the values in the desired format.
